### ISSUE TITLE
Possible package update

### ISSUE THREAD

**Issue by telesvar** [puyuan/py-nanoid#13]
Hello, as a prerequisite, I want to tell you that this post falls in a category 'I wonder, what if...'.
So, I just want to leave you thinking about making this package even smaller and in one file.

Reasons are:
1. Make package more maintainable
2. The short declaration does not make functions significantly slower

Drawbacks:
1. `secrets` package is not available in Python 2 👎 

In the end, this what makes Python really stand out: short but clear expressions, which are easy to read and write.

```python
from functools import reduce
from operator import concat

import random
import secrets

alphabet_std = '_-0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'
default_size = 21


def non_secure_generate(alphabet=alphabet_std, size=default_size):    
    return reduce(concat, (random.choice(alphabet) for _ in range(size)))


def generate(alphabet=alphabet_std, size=default_size):
    return reduce(concat, (secrets.choice(alphabet) for _ in range(size)))
```

**Comment by telesvar** [comment]
Also, I am aware that I have removed genuine generation algorithm from `generate`, but a random generator in Python is far better than in Node.js, therefore, the sacrifice.

**Comment by puyuan** [comment]
Looks quite elegant to me.  I am just wondering for correctness, if  your latest  version is equivalent to  the algorithm version?  Given its written very differently, I am not able to tell. 

Also due to its conciseness,, if there is a change in javascript version, it leaves no room for us to make equivalent changes.

**Comment by telesvar** [comment]
You mean usage of urandom as cryptographically safe randomness? Yes, `secrets` relies on `random.SystemRandom`, which is inherently urandom.

Nothing new in JS world this package could have any benefits of.

**Comment by telesvar** [comment]
You can check out, how the variant works in my repo https://github.com/aidarkhanov/py-nanoid

**Comment by puyuan** [comment]
> You mean usage of urandom as cryptographically safe randomness? Yes, `secrets` relies on `random.SystemRandom`, which is inherently urandom.
> 
> Nothing new in JS world this package could have any benefits of.

Oh, I meant the original js algorithm has all the masks and bitshifts. Now, your implementation is greatly simplified, is it equivalent?

**Comment by telesvar** [comment]
From [what tests can say about distribution](https://github.com/aidarkhanov/py-nanoid/blob/master/test/generate_test.py#L19) algorithm equivalency, yes.

**Comment by TrippleA-Ashaba** [comment]
Hello, was this implementation considered? Using [`secrets`](https://docs.python.org/3/library/secrets.html#module-secrets) is a much better implementation and comes with the standard library. 
Although I have a few suggestions on the implementation. use `join` instead of `concat` or `+=`


```
import secrets
import string

alphabet_std = string.ascii_lowercase + string.digits


def generate(alphabet=alphabet_std, size=21) -> str:
    return "".join(secrets.choice(alphabet) for _ in range(size))
```

**Comment by telesvar** [comment]
@TrippleA-Ashaba this library was created before secrets had choice :) I am not programming in Python anymore, so I cannot guarantee that the choice method has uniform randomness as in the library's implementation. If secrets is as good then I do not see why you should use py-nanoid ;) Maybe for a symmetry with the original JS package.

**Comment by TrippleA-Ashaba** [comment]
@aidarkhanov you are right you can just use the secrets module directly out of the box. I will make more research on the topic. As of now `secrets.choice` or `secrets.token_urlsafe` are good alternatives to UUID for my use case. 
thanks.
